velox/functions/prestosql/window/LeadLag.cpp

```diff
@@ -286,6 +291,11 @@ void LeadLagFunction<true>::setRowNumbersForConstantOffset() {
   }
 
   auto constantOffsetValue = constantOffset_.value();
+  // Set row number to kNullRow for out of range offset.
```

Comment:
There is a good amount of copy-paste between the two specializations of :setRowNumbersForConstantOffset now. It would be nice to extract the logic to handle null and out-of-bound offset. Maybe,

```
void LeadLagFunction::setRowNumbersForConstantOffset() {
  if (isConstantOffsetNull_) {
    std::fill(rowNumbers_.begin(), rowNumbers_.end(), kNullRow);
    return;
  }

  auto constantOffsetValue = constantOffset_.value();
  // Set row number to kNullRow for out of range offset.
  if (constantOffsetValue > partition_->numRows()) {
    std::fill(rowNumbers_.begin(), rowNumbers_.end(), kNullRow);
    return;
  }

  setRowNumbersForConstantOffset(constantOffsetValue);

template <>
void LeadLagFunction<true>::setRowNumbersForConstantOffset(vector_size_t offset);

template <>
void LeadLagFunction<false>::setRowNumbersForConstantOffset(vector_size_t offset);
}
```